logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/user/SetUserHandleUseCaseTest.kt

```diff
@@ -56,6 +69,82 @@ class SetUserHandleUseCaseTest {
         verify(userRepository)
             .coroutine { updateSelfHandle(handle) }
             .wasInvoked(exactly = once)
+        verify(userRepository)
+            .coroutine { updateLocalSelfUserHandle(handle) }
+            .wasInvoked(exactly = once)
+    }
+
+    @Test
+    fun givenValidHandleAndRepositorySuccess_whenSlowSyncIsOngoing_thenLocalDataUpdatedAfterSlowSyncAndSuccessIsPropagated() = runTest {
+        val handle = "user_handle"
+        given(validateHandleUseCase)
+            .function(validateHandleUseCase::invoke)
+            .whenInvokedWith(any())
+            .then { true }
+        given(userRepository)
+            .coroutine { updateSelfHandle(handle) }
+            .then { Either.Right(Unit) }
+        given(userRepository)
+            .coroutine { updateLocalSelfUserHandle(handle) }
+            .then { }
+        given(syncManager)
+            .coroutine { syncManager.isSlowSyncOngoing() }
+            .then { true }
+        given(syncManager)
+            .coroutine { syncManager.isSlowSyncCompleted() }
+            .then { true }
+        given(syncManager)
+            .coroutine { syncManager.waitForSlowSyncToComplete() }
+            .then { }
```

Comment:
Same here:

```suggestion```

You can delete stubs of `Unit` functions if you define the Mock like this:

```kotlin
    @Mock
    private val syncManager = configure(mock(classOf<SyncManager>())) { stubsUnitByDefault = true }
```